### PR TITLE
2 packages from c-cube/tiny_httpd at 0.7

### PR DESCRIPTION
**Pull request by c-cube**
This pull-request concerns:
-`tiny_httpd.0.7`: Minimal HTTP server using good old threads
-`tiny_httpd_camlzip.0.7`: Interface to camlzip for tiny_httpd

---
* Homepage: https://github.com/c-cube/tiny_httpd/
* Source repo: git+https://github.com/c-cube/tiny_httpd.git
* Bug tracker: https://github.com/c-cube/tiny_httpd/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

**Comment by mseri** [comment]
Only failure is a failure running tests on tiny_httpd with ocaml 4.12:
```
#=== ERROR while compiling tiny_httpd.0.7 =====================================#
# context              2.0.7 | linux/x86_64 | ocaml-variants.4.12.0+trunk | pinned(https://github.com/c-cube/tiny_httpd/archive/0.7.tar.gz)
# path                 ~/.opam/4.12/.opam-switch/build/tiny_httpd.0.7
# command              ~/.opam/4.12/bin/dune runtest -p tiny_httpd
# exit-code            1
# env-file             ~/.opam/log/tiny_httpd-29-bd50e7.env
# output-file          ~/.opam/log/tiny_httpd-29-bd50e7.out
### output ###
#       ocamlc src/.tiny_httpd.inline-tests/.tiny_httpd.inline-tests.eobjs/byte/dune__exe__Inline_test_runner_tiny_httpd.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -24 -g -bin-annot -I src/.tiny_httpd.inline-tests/.tiny_httpd.inline-tests.eobjs/byte -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/oUnit -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/qcheck -I src/.tiny_httpd.objs/byte -I src/.tiny_httpd.objs/native -no-alias-deps -o src/.tiny_httpd.inline-tests/.tiny_httpd.inline-tests.eobjs/byte/dune__exe__Inline_test_runner_tiny_httpd.cmo -c -impl src/.tiny_httpd.inline-tests/inline_test_runner_tiny_httpd.ml-gen)
# File "Tiny_httpd.ml", line 572, characters 6-22:
# Error: Unbound module QCheck_ounit

-       ocamlc src/.tiny_httpd.inline-tests/.tiny_httpd.inline-tests.eobjs/byte/dune__exe__Inline_test_runner_tiny_httpd.{cmi,cmo,cmt} (exit 2)
- (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -24 -g -bin-annot -I src/.tiny_httpd.inline-tests/.tiny_httpd.inline-tests.eobjs/byte -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/oUnit -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/qcheck -I src/.tiny_httpd.objs/byte -I src/.tiny_httpd.objs/native -no-alias-deps -o src/.tiny_httpd.inline-tests/.tiny_httpd.inline-tests.eobjs/byte/dune__exe__Inline_test_runner_tiny_httpd.cmo -c -impl src/.tiny_httpd.inline-tests/inline_test_runner_tiny_httpd.ml-gen)
- File "Tiny_httpd.ml", line 572, characters 6-22:
- Error: Unbound module QCheck_ounit
```

For me, we can just disable the tests with that version. Let me know if you prefer something else

File: packages/tiny_httpd/tiny_httpd.0.7/opam [hunk 1]
```diff
@@ -0,0 +1,31 @@
+opam-version: "2.0"
+authors: ["Simon Cruanes"]
+maintainer: "simon.cruanes.2007@m4x.org"
+license: "MIT"
+synopsis: "Minimal HTTP server using good old threads"
+build: [
+  ["dune" "build" "@install" "-p" name "-j" jobs]
+  ["dune" "build" "@doc" "-p" name] {with-doc}
+  ["dune" "runtest" "-p" name] {with-test}
```
Comment (mseri) [inline packages/tiny_httpd/tiny_httpd.0.7/opam:9]:
```suggestion
  ["dune" "runtest" "-p" name] {with-test & ocaml:version < "4.12"}
```

Comment (kit-ty-kate) [reply packages/tiny_httpd/tiny_httpd.0.7/opam:9]: The error is due to a simple missing constraint. I would prefer that over disabling the tests. I'm on it.

**Comment by kit-ty-kate** [comment]
All green now with the fix. @c-cube could you return it in the opam file upstream so that this doesn't happen next time?

Thanks!

File: packages/tiny_httpd/tiny_httpd.0.7/opam (hunk 1 above)
Comment (c-cube) [reply packages/tiny_httpd/tiny_httpd.0.7/opam:9]: is there a reason why the test fails on 4.12? I can't think of anything in qcheck_ounit that requires < 4.12?

Comment (c-cube) [reply packages/tiny_httpd/tiny_httpd.0.7/opam:9]: I'm not using stdlib-shims anywhere, I suppose it's a transitive dep that does?

Comment (kit-ty-kate) [reply packages/tiny_httpd/tiny_httpd.0.7/opam:9]: Yes, in `ounit2`

Comment (c-cube) [reply packages/tiny_httpd/tiny_httpd.0.7/opam:9]: seems to me that it's either a bug in opam or ounit2's package or travis then. I expect that if my test deps are not compatible with 4.12, then opam should know it's not possible to run tests? 

In other words (sorry, I'm tired): opam should know that the test don't run, imho. I should not have to write constraints that are already expressed in my transitive dependencies.

**Pull request merged**